src/filedata.h

```diff
@@ -132,8 +132,182 @@ struct FileData {
 
 	gint page_num;
 	gint page_total;
+
+	static gchar *text_from_size(gint64 size);
+	static gchar *text_from_size_abrev(gint64 size);
+	static const gchar *text_from_time(time_t t);
+
+	/**
+	 * @headerfile file_data_new_group
+	 * scan for sidecar files - expensive
+	 */
+	static FileData *file_data_new_group(const gchar *path_utf8);
+
+	/**
+	 * @headerfile file_data_new_no_grouping
+	 * should be used on helper files which can't have sidecars
+	 */
+	static FileData *file_data_new_no_grouping(const gchar *path_utf8);
+
+	/**
+	 * @headerfile file_data_new_dir
+	 * should be used on dirs
+	 */
+	static FileData *file_data_new_dir(const gchar *path_utf8);
+
+	static FileData *file_data_new_simple(const gchar *path_utf8);
+
+#ifdef DEBUG_FILEDATA
+	FileData *file_data_ref(const gchar *file = __builtin_FILE(), gint line = __builtin_LINE());
+	void file_data_unref(const gchar *file = __builtin_FILE(), gint line = __builtin_LINE());
+#else
+	FileData *file_data_ref();
+	void file_data_unref();
+#endif
+
+	void file_data_lock(FileData *fd);
```

Comment:
Is removing `FileData` parameter from non-static methods planned in the next steps?

Reply:
Yeah.  Going to drop that, and also drop the C-style "file_data_" name prefix.  So the end state for this method is `void lock();`